pkg/server/server_test.go

```diff
@@ -0,0 +1,123 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"testing"
+
+	"github.com/ipfs/testground/pkg/daemon/client"
+)
+
+var srv *Server
```

Comment:
I think it'd be better to not have global vars in tests. We want to be able to run parallel tests. Also the port of the server should be given by the kernel, so just us ":0", and not a hardcoded one.

Reply:
Doooone!